### ISSUE TITLE
typescript definition does not match code for mssql domain

### ISSUE THREAD

**Issue by shaungreen** [knex/knex#3116]
# Environment

Knex version: 0.16.3
Database + version: mssql (any version)
OS: (any)

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

Typescript type Knex.MsSqlConnectionConfig specifies that the 'domain' attribute should be found under the 'options' object, but the code in src/dialects/mssql/index.js (line 97) expects the 'domain' attribute to be found as a top-level attribute of the config.

Work-around is to include a typescript exception in the code when creating the knex config object:

```
const mssqlConfig = (
  credentials: UsernamePassword
): Knex.MsSqlConnectionConfig => ({
  database: "myDbName",
  server: "myServerName",
  user: credentials.username,
  password: credentials.password,
  // @ts-ignore: knex type is mismatched with code
  domain: "myDomain"
})
```

2. Error message

Typescript compile error if @ts-ignore option not used.

**Comment by kibertoad** [comment]
@shaungreen Would you consider submitting a PR to fix this issue?

**Comment by shaungreen** [comment]
No problem - what branch should I use?

**Comment by kibertoad** [comment]
@shaungreen Could you please open PR against our repo with the fix?

**Comment by windupbird144** [comment]
Is this issue still open? It looks to me like it was fixed and merged here: https://github.com/knex/knex/pull/3269

**Comment by OlivierCavadenti** [comment]
> Is this issue supposed to be open? It looks to me like it was fixed and merged here: #3269

Good catch, I close it.
